src/corppa/ocr/gvision_ocr.py

```diff
@@ -7,60 +7,168 @@
+            image = vision.Image(content=content)
+
+            # Performs OCR and handwriting detection on the image file
+            response = client.document_text_detection(image=image)
+
+            # Save plain text output to local file;
+            # even if text is empty, create text file so we don't request again
+            with open(textfile, "w") as textfilehandle:
+                textfilehandle.write(response.full_text_annotation.text)
+
+            # Save json responst
+            json_response = vision.AnnotateImageResponse.to_json(response)
+            with open(jsonfile, "w") as jsonfilehandle:
+                jsonfilehandle.write(json_response)
+
+            if response.error.message:
+                raise Exception(
+                    f"{response.error.message}\n for more info on error messages, "
+                    "check: https://cloud.google.com/apis/design/errors"
                 )
-            )
 
-        # stop processing after first image until we make decisions & refine
-        break
+            # Update counter
+            ocr_count += 1
+            if show_progress:
+                progress_bar.update()
+
+            # Check if we should stop
+            if ocr_limit and ocr_count == ocr_limit:
+                # TODO: Is there a better structuring to avoid this break
+                break
+
+    if show_progress:
+        # Close progress bar
+        progress_bar.close()
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="Uses Google Vision API to OCR images."
+    )
+
+    # Required arguments
+    parser.add_argument(
+        "input",
+        help="Top-level input directory containing images to be OCR'd",
+        type=pathlib.Path,
+    )
+    parser.add_argument(
+        "output",
+        help="Top-level output directory for OCR output; "
+        + "maintains input subdirectory structure.",
+        type=pathlib.Path,
+    )
+
+    # Optional arguments
+    parser.add_argument(
+        "--progress",
+        help="Show progress",
+        action=argparse.BooleanOptionalAction,
+        default=True,
+    )
+    parser.add_argument(
+        "--ocr-limit",
+        help="Set a limit for the number of images to be OCR'd",
+        type=int,
+        default=0,
+    )
+    parser.add_argument(
+        "--ext",
+        help="Accepted file extension(s). Defaults: .TIF, .jpg",
+        nargs="*",
+        type=str,
+        action="extend",
+    )
+
+    args = parser.parse_args()
+    # Workaround: Set default extensions if none are provided.
+    if args.ext is None:
+        args.ext = [".TIF", ".jpg"]
+
+    # Validate arguments
+    if not args.input.is_dir():
+        print(f"Error: input directory {args.input} does not exist")
+        sys.exit(1)
+    # TODO: Is this too restrictive / unnecessary?
+    if not args.output.is_dir():
+        print(f"Error: output directory {args.output} does not exist")
+        sys.exit(1)
```

Comment:
might be nice to have an option to create it for you but I'd rather be warned than have it happily put things somewhere I didn't intend!

Reply:
I think it depends on our expected use case, since this is the top-level directory. It seems important that it exists to avoid the mistake where the output directory is wrong for some reason so any previous OCR efforts cannot be found and will be fully duplicated in the "new" output directory.